src/services/sourcemaps.ts

```diff
@@ -0,0 +1,330 @@
+            const char = state.encodedText.charAt(state.decodingIndex);
+            if (char === ";") {
+                // New line
+                state.currentEmittedLine++;
+                state.currentEmittedColumn = 1;
+                state.decodingIndex++;
+                continue;
+            }
+
+            if (char === ",") {
+                // Next entry is on same line - no action needed
+                state.decodingIndex++;
+                continue;
+            }
+
+            // Read the current span
+            // 1. Column offset from prev read jsColumn
+            state.currentEmittedColumn += base64VLQFormatDecode();
+            // Incorrect emittedColumn dont support this map
+            if (createErrorIfCondition(state.currentEmittedColumn < 1, "Invalid emittedColumn found")) {
+                return;
+            }
+            // Dont support reading mappings that dont have information about original source and its line numbers
+            if (createErrorIfCondition(isSourceMappingSegmentEnd(state.encodedText, state.decodingIndex), "Unsupported Error Format: No entries after emitted column")) {
+                return;
+            }
+
+            // 2. Relative sourceIndex
+            state.currentSourceIndex += base64VLQFormatDecode();
+            // Incorrect sourceIndex dont support this map
+            if (createErrorIfCondition(state.currentSourceIndex < 0, "Invalid sourceIndex found")) {
+                return;
+            }
+            // Dont support reading mappings that dont have information about original source span
+            if (createErrorIfCondition(isSourceMappingSegmentEnd(state.encodedText, state.decodingIndex), "Unsupported Error Format: No entries after sourceIndex")) {
+                return;
+            }
+
+            // 3. Relative sourceLine 0 based
+            state.currentSourceLine += base64VLQFormatDecode();
+            // Incorrect sourceLine dont support this map
+            if (createErrorIfCondition(state.currentSourceLine < 1, "Invalid sourceLine found")) {
+                return;
+            }
+            // Dont support reading mappings that dont have information about original source and its line numbers
+            if (createErrorIfCondition(isSourceMappingSegmentEnd(state.encodedText, state.decodingIndex), "Unsupported Error Format: No entries after emitted Line")) {
+                return;
+            }
+
+            // 4. Relative sourceColumn 0 based
+            state.currentSourceColumn += base64VLQFormatDecode();
+            // Incorrect sourceColumn dont support this map
+            if (createErrorIfCondition(state.currentSourceColumn < 1, "Invalid sourceLine found")) {
+                return;
+            }
+            // 5. Check if there is name:
+            if (!isSourceMappingSegmentEnd(state.encodedText, state.decodingIndex)) {
+                if (state.currentNameIndex === undefined) {

```

Comment:
We don't use names, so should we care about the name index or whether it's invalid? We could just as easily advance the position until we hit the segment end. I'd rather air on the side of leniency for a better user experience.